packages/salesforcedx-vscode-lwc/test/vscode-integration/jsconfig.test.ts

```diff
@@ -0,0 +1,215 @@
+  });
+
+  it('Should not update jsconfig.json on keystrokes', async function() {
+    this.timeout(5000);
+
+    const document = await workspace.openTextDocument(
+      path.join(modulesDir, TEST_COMPONENT_NAME, `${TEST_COMPONENT_NAME}.js`)
+    );
+    const editor = await window.showTextDocument(document);
+
+    await editor.edit(editBuilder => {
+      editBuilder.insert(new Position(0, 0), 'foo');
+    });
+
+    const didUpdate = await waitForConfigUpdate(configPath);
+    expect(didUpdate).to.equal(false);
+
+    const newConfig = await parseConfig(configPath);
+    expect(config).to.eql(newConfig);
+  });
+
+  afterEach(async function() {
+    this.timeout(5000);
+
+    await workspace.fs.delete(
+      URI.file(path.join(modulesDir, TEST_COMPONENT_NAME)),
+      { recursive: true }
+    );
+    await workspace.fs.delete(
+      URI.file(path.join(modulesDir, CREATE_COMPONENT_NAME)),
+      { recursive: true }
+    );
+
+    await waitForConfigUpdate(configPath);
+  });
+});
+
+/**
+ * Helper to read and parse jsconfig.json
+ *
+ * @param configPath - path to jsconfig.json
+ */
+async function parseConfig(configPath: string) {
+  return JSON.parse(
+    (await workspace.fs.readFile(URI.file(configPath))).toString()
+  );
+}
+
+/**
+ * Helper to wait for jsconfig.json updates
+ *
+ * @param configPath - path to jsconfig.json
+ */
+async function waitForConfigUpdate(configPath: string): Promise<boolean> {
+  return await new Promise<boolean>(resolve => {
+    let count = 0;
+    const watcher = fs.watch(configPath, event => {
+      if (event === 'change') {
+        count++;
+
+        // both language servers update the jsconfig file, so we must wait for
+        // for both to write to the config file.
+        if (count === 2) {
+          watcher.close();
+          resolve(true);
+        }
+      }
+    });
+
+    setTimeout(() => {
+      watcher.close();
+      resolve(false);
+    }, 2000);
+  });
+}
+
+/**
+ * Helper to create a LWC component
+ * Ideally, this should be replaced with invoking `sfdx.force.lightning.lwc.create`
+ *
+ * @param name - name of the new component
+ * @param modulesDir - path for where to create the component
+ */
+async function createComponent(name: string, modulesDir: string) {
+  await workspace.fs.createDirectory(URI.file(path.join(modulesDir, name)));
+
+  const htmlEdit = new WorkspaceEdit();
+  const jsEdit = new WorkspaceEdit();
+  const xmlEdit = new WorkspaceEdit();
+
+  htmlEdit.createFile(URI.file(path.join(modulesDir, name, `${name}.html`)));
+  jsEdit.createFile(URI.file(path.join(modulesDir, name, `${name}.js`)));
+  xmlEdit.createFile(URI.file(path.join(modulesDir, name, `${name}-meta.xml`)));
+
+  await workspace.applyEdit(htmlEdit);
+  await workspace.applyEdit(jsEdit);
+  await workspace.applyEdit(xmlEdit);
+
+  const encoder = new TextEncoder();
+
+  await workspace.fs.writeFile(
+    URI.file(path.join(modulesDir, name, `${name}.html`)),
+    encoder.encode(`<template>
```

Comment:
Small nit, not worth changing if you don't agree. These code snippets could benefit by being delcared at the top of the file. That separates the declaration of data from the processing. I think 99% of the time, we'll want to tweak the content only.